extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesUtil.java

```diff
@@ -107,6 +107,20 @@ static ReadOptionalResponse createReadOptionalValueAndConvertIfNeeded(String pro
         return new ReadOptionalResponse(value, isPresentTrue, isPresentBranch.falseBranch());
     }
 
+    public static boolean isListOfObject(Type type) {
+        if (type.kind() != Type.Kind.PARAMETERIZED_TYPE) {
+            return false;
+        }
+        ParameterizedType parameterizedType = (ParameterizedType) type;
+        if (!DotNames.LIST.equals(parameterizedType.name())) {
+            return false;
+        }
+        if (parameterizedType.arguments().size() != 1) {
+            return false;
+        }
+        return !parameterizedType.arguments().get(0).name().toString().startsWith("java");
```

Comment:
`startsWith("java.util.List")`?

Reply:
Not really, because here we are looking for types like Integer, String, Long and the like